### PR TITLE
kiss: replace stat usage with portable parsing of ls -l

### PR DESCRIPTION
**Pull request by E5ten**

### CODE REVIEW

File: kiss [hunk 1]
```diff
@@ -1163,7 +1174,8 @@ pkg_updates() {
                 # ownership of files and directories in the rare
                 # case that the repository is owned by a 3rd user.
                 (
-                    user=$(stat -c %U "$PWD") || user=root
+                    user=$(ls -ld "$PWD") || user=root
+                    user=${user#* * } user=${user%% *}
```
Comment (dylanaraps) [inline kiss:1178]: I've implemented the user calls here: https://github.com/kisslinux/kiss/commit/278c222902a38046012fdfc94e40269bd3378a9b#diff-4188679c1d8a284ccc41a6b601869e05 :+1:

File: kiss [hunk 2]
```diff
@@ -818,7 +818,18 @@ pkg_install_files() {
...
+        for c in 1 2 3 4 5 6 7 8 9; do
+            out=${out#?}
+            case $out in
+                [rwx]*) : "$((o += c % 3 ? c % 3 % 2 ? 4 : 2 : 1))" ;;
+                [ST]*) : "$((b += c > 3 ? c > 6 ? 1 : 2 : 4))" ;;
+                [st]*) : "$((b += c > 3 ? c > 6 ? 1 : 2 : 4))" "$((o += 1))" ;;
+            esac
+            [ "$((c % 3))" -eq 0 ] && rwx=$rwx$o o=0
+        done
+        rwx=$b$rwx
```
Comment (dylanaraps) [inline kiss:832]: I get `/home/dylan/bin/p2o: line 26: malformed ?: operator` when trying in `ash` with any file.

File: kiss (hunk 1 above)
Comment (E5ten) [reply kiss]: I've rebased on that commit

File: kiss (hunk 2 above)
Comment (E5ten) [reply kiss]: I put parentheses around the inner ternary operator usages, it seems to work with ash now.

**Comment by dylanaraps** [comment]
I've committed an alternative version here: https://github.com/kisslinux/kiss/commit/d0100b03a113cc7bc7b9845667c43ef8dabcad60
 
Thanks!

**Comment by dylanaraps** [comment]
Let me know if you have any further suggestions. Bye bye `stat`. :D

**Pull request closed without merging**